tensorflow/tools/docker/Dockerfile.devel-gpu-cuda9-cudnn7

```diff
@@ -79,23 +80,43 @@ RUN git clone https://github.com/tensorflow/tensorflow.git && \
     git checkout ${TF_BRANCH}
 WORKDIR /tensorflow
 
-# Configure the build for our CUDA configuration.
-ENV CI_BUILD_PYTHON python
-ENV LD_LIBRARY_PATH /usr/local/cuda/extras/CUPTI/lib64:$LD_LIBRARY_PATH
-ENV TF_NEED_CUDA 1
-ENV TF_CUDA_COMPUTE_CAPABILITIES 3.0,3.5,5.2,6.0,6.1
-ENV TF_CUDA_VERSION 9.0
-ENV TF_CUDNN_VERSION 7
-RUN ./configure
-
-RUN LD_LIBRARY_PATH=/usr/local/cuda/lib64/stubs:${LD_LIBRARY_PATH} \
-    bazel build -c opt --config=cuda --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0" \
-        --jobs=${TF_AVAILABLE_CPUS} \
-        tensorflow/tools/pip_package:build_pip_package && \
-    mkdir -p /pip_pkg && \
+# Tell TensorFlow how to configure the build, in detail.
+ENV CI_BUILD_PYTHON=python \
+    LD_LIBRARY_PATH=/usr/local/cuda/extras/CUPTI/lib64:${LD_LIBRARY_PATH} \
+    CUDNN_INSTALL_PATH=/usr/lib/x86_64-linux-gnu \
+    PYTHON_BIN_PATH=/usr/bin/python \
+    PYTHON_LIB_PATH=/usr/local/lib/python2.7/dist-packages \
+    TF_NEED_CUDA=1 \
+    TF_CUDA_VERSION=9.0 \
+    TF_CUDA_COMPUTE_CAPABILITIES=3.0,3.5,5.2,6.0,6.1,7.0 \
+    TF_CUDNN_VERSION=7 \
+    TF_NEED_JEMALLOC=1 \
+    TF_NEED_GCP=0 \
+    TF_NEED_HDFS=0 \
+    TF_ENABLE_XLA=1 \
+    TF_NEED_GDR=0 \
+    TF_NEED_VERBS=0 \
+    TF_NEED_OPENCL=0 \
+    TF_NEED_S3=0 \
+    TF_CUDA_CLANG=0 \
+    TF_NEED_MPI=0
+
+# Configure, Build and Install TensorFlow.
+RUN ln -s /usr/local/cuda/lib64/stubs/libcuda.so /usr/local/cuda/lib64/stubs/libcuda.so.1 && \
```

Comment:
No need for this.
You need to use `nvidia-docker` which makes the `ibcuda.so.1` link available.

Reply:
I am not sure where the ```libcuda.so.1``` link is, as you can see:
```
chweng@honghuTX:/dev/shm/tmp$ nvidia-docker build -t 'test' .
Sending build context to Docker daemon  221.4MB
Step 1/3 : FROM nvidia/cuda:9.0-cudnn7-devel-ubuntu16.04
 ---> 86c4dcdd14b5
Step 2/3 : RUN ls -hl  /usr/local/cuda/lib64/stubs/libcuda*
 ---> Running in 7234e9e1d37b
-rw-r--r-- 1 root root 42K Sep  2 10:39 /usr/local/cuda/lib64/stubs/libcuda.so
 ---> ba363cc03ba6
Removing intermediate container 7234e9e1d37b
Step 3/3 : RUN ls -hl  /usr/local/cuda-9.0/targets/x86_64-linux/lib/stubs/libcuda*
 ---> Running in 9d99b95ea77b
-rw-r--r-- 1 root root 42K Sep  2 10:39 /usr/local/cuda-9.0/targets/x86_64-linux/lib/stubs/libcuda.so
 ---> e073112d01a3
Removing intermediate container 9d99b95ea77b
Successfully built e073112d01a3
Successfully tagged test:latest
```
That is to say, you can find ```libcuda.so``` in  ```/usr/local/cuda-9.0/targets/x86_64-linux/lib/stubs/``` or ```/usr/local/cuda-9.0/targets/x86_64-linux/lib/stubs/```, but I don't know where to look for the ```libcuda.so.1``` link.

Reply:
@flx42 to guide us here.

Reply:
During `build`, nvidia-docker doesn't do anything (i.e. you can't have GPU access). You told me this line is to workaround a problem you had at build time:
https://github.com/tensorflow/tensorflow/pull/13399#issuecomment-333637487

But it looks like the line afterwards is missing an `export`, otherwise `LD_LIBRARY_PATH` won't be available for the configure script.

Reply:
Now I remember, thanks for reminding me.
@chi-hung please add `export` before the variable definition in the next line as instructed by @flx42

Reply:
I think it's no need to export that temporary variable. if you have a script called *test.sh*:
```
#!/bin/bash
echo ${my_var}
```
which does nothing but print the variable *my_var*, and you execute this script in this way:
```
my_var=100 ./test.sh
```
the output will be 100 and *my_var* will be known (and only be known) to this script. See the following quick test:
```
chweng@honghuTX:/dev/shm/tmp$ my_var=100 ./test.sh
100
chweng@honghuTX:/dev/shm/tmp$ env | grep my_var
chweng@honghuTX:/dev/shm/tmp$
```

Reply:
My bad, I didn't see there was no `&&` at the end of this `LD_LIBRARY_PATH` line. Visually it's a bit difficult to follow.